Sources/Structs/Weak.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  Created by Murat Yilmaz on 19.05.18.
+//  Copyright © 2018 Flinesoft. All rights reserved.
+//
+
+import Foundation
+
+/// A wrapper for storing weak references to a `Wrapped` instance.
+public struct Weak<Wrapped>: ExpressibleByNilLiteral where Wrapped: AnyObject {
+    /// The value of `Wrapped` stored as weak reference
+    public weak var value: Wrapped?
```

Comment:
Maybe we should name this `weakValue` instead so the user knows immediately without looking at the type that it's weak?

Reply:
I wouldn't recommend an annotated identifier, consider refactorability. 

By keeping a more unified interface you can easily refactor a Weak wrapper to an Unowned one without further effort on rewriting the property names.

Reply:
Agreed, let's keep it then.